test/torchaudio_unittest/datasets/musdb_hq_test.py

```diff
@@ -0,0 +1,164 @@
+import os
+
+import torch
+from parameterized import parameterized
+from torchaudio.datasets import musdb_hq
+from torchaudio.datasets.musdb_hq import _VALIDATION_SET
+from torchaudio_unittest.common_utils import (
+    get_whitenoise,
+    save_wav,
+    TempDirMixin,
+    TorchaudioTestCase,
+)
+
+_SOURCE_SETS = [
+    (None,),
+    (["bass", "drums", "other", "vocals"],),
+    (["bass", "drums", "other"],),
+    (["bass", "drums", "vocals"],),
+    (["bass", "vocals", "other"],),
+    (["vocals", "drums", "other"],),
+    (["mixture"],),
+]
+
+seed_dict = {
+    "bass": 0,
+    "drums": 1,
+    "other": 2,
+    "mixture": 3,
+    "vocals": 4,
+}
+
+EXT = ".wav"
+
+
+def _save_sample(dataset_dir, folder, song, source, sample_rate, seed):
+    # create and save audio samples to corresponding files
+    path = os.path.join(dataset_dir, folder)
+    os.makedirs(path, exist_ok=True)
+    song_path = os.path.join(path, str(song))
+    os.makedirs(song_path, exist_ok=True)
+    source_path = os.path.join(song_path, f"{source}{EXT}")
+
+    data = get_whitenoise(
+        sample_rate=sample_rate,
+        duration=5,
+        n_channels=2,
+        seed=seed,
+    )
+    save_wav(source_path, data, sample_rate)
+
+    sample = (data, sample_rate, 5 * sample_rate, song)
+
+    return sample
+
+
+def _get_mocked_samples(dataset_dir, sample_rate):
+    sample_count = 5
+
+    all_samples = {"train": {}, "test": {}}
+
+    folders = ["train", "test"]
+    sources = ["bass", "drums", "other", "mixture", "vocals"]
+
+    curr_idx = 0
+    for folder in folders:
+        for _ in range(sample_count):
+            sample_list = []
+            for source in sources:
+                sample = _save_sample(dataset_dir, folder, str(curr_idx), source, sample_rate, seed_dict.get(source))
+                sample_list.append(sample)
+            all_samples[folder][str(curr_idx)] = sample_list
+            curr_idx += 1
+        if folder == "train":
+            _VALIDATION_SET.sort()
```

Comment:
global variables shouldn't be modified. also this line isn't necessary here since in the dataset, the names are only sorted if the split isn't the valid set